src/simtools/testing/sim_telarray_metadata.py

```diff
@@ -0,0 +1,164 @@
+"""Test consistency of sim_telarray metadata."""
+
+import logging
+
+import numpy as np
+
+from simtools.simtel.simtel_config_reader import SimtelConfigReader
+from simtools.simtel.simtel_io_metadata import (
+    get_sim_telarray_telescope_id,
+    read_sim_telarray_metadata,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def assert_sim_telarray_metadata(file, array_model):
+    """
+    Assert consistency of sim_telarray metadata with given array model.
+
+    Parameters
+    ----------
+    file: Path
+        Path to the sim_telarray file.
+    array_model: ArrayModel
+        Array model to compare with.
+    """
+    global_meta, telescope_meta = read_sim_telarray_metadata(file)
+    _logger.info(f"Found metadata in sim_telarray file for {len(telescope_meta)} telescopes")
+    site_parameter_mismatch = _assert_model_parameters(global_meta, array_model.site_model)
+
+    if len(telescope_meta) != len(array_model.telescope_model):
+        raise ValueError(
+            f"Number of telescopes in sim_telarray file ({len(telescope_meta)}) does not match "
+            f"number of telescopes in array model ({len(array_model.telescope_model)})"
+        )
+
+    for telescope_name in array_model.telescope_model.keys():
+        if not get_sim_telarray_telescope_id(telescope_name, file):
+            raise ValueError(f"Telescope {telescope_name} not found in sim_telarray file metadata")
+
+    telescope_parameter_mismatch = [
+        _assert_model_parameters(telescope_meta[i], model)
+        for i, (_, model) in enumerate(array_model.telescope_model.items(), start=1)
+    ]
+
+    # ensure printout of all mismatches, not only those found first
```

Comment:
This comment is true for the second `if` statement, but if there's a mismatch in the site parameters, we won't reach the telescope parameters. Either we collect all errors and raise the ValueError afterwards, or perhaps join the mismatch lists so the `any` applies to the site one as well? Otherwise I don't mind if you simply want to move this comment to above the telescope mismatch check, at least there it will be correct.

Reply:
I am aware of that. Your suggestion is good to merge the site / telescope mismatches and then check them. This allows to print all errors.